.github/workflows/zxc-build-publish-state-validator.yaml

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: Apache-2.0
+name: "ZXC: Build & Publish Hedera State Validator Uber JAR"
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: "Branch, tag, or commit to build"
+        required: false
+        type: string
+  workflow_call:
+    inputs:
+      ref:
+        description: "Branch, tag, or commit to build"
+        required: false
+        type: string
+    outputs:
+      gcs-uri:
+        description: "GCS URI of the published uber JAR"
+        value: ${{ jobs.build-publish.outputs.gcs-uri }}
+
+permissions:
+  contents: read
+  id-token: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-publish:
+    name: Build & Publish State Validator Uber JAR
+    runs-on: hiero-citr-linux-medium
+
+    outputs:
+      gcs-uri: ${{ steps.upload.outputs.gcs_uri || '' }}
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@ec9f2d5744a09debf3a187a3f4f675c53b671911 # v2.13.0
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.ref || github.sha }}
+
+      - name: Set up Java
+        uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4.7.1
+        with:
+          distribution: temurin
+          java-version: "21"
+
+      - name: Set up Gradle
+        uses: gradle/actions/setup-gradle@d156388eb19639ec20ade50009f3d1993974129d # v4.3.0
+        with:
+          gradle-home-cache-cleanup: true
+
+      - name: Build State Validator Uber shadowJar
+        run: |
+          ./gradlew --no-daemon --stacktrace :hedera-state-validator:shadowJar
+
+      - name: Determine version and artifact
+        id: version-and-artifact
+        run: |
+          set -euo pipefail
+          ROOT_VER_FILE="version.txt"
```

Comment:
Just use the gradle step which calculates the version number and the semver command if you need to parse it.